### ISSUE TITLE
Firehose stops forwarding metrics to nozzle

### ISSUE THREAD

**Issue by vikramraman** [cloudfoundry/cf-java-client#928]
We are encountering an issue where our Java based Nozzle stops receiving any data from the firehose about ~40 seconds after starting up. We do not see any clues in the logs, but our nozzle does report internal metrics that indicate this scenario.

The logs for reference:

```
   2018-12-11T23:20:32.88+0000 [APP/PROC/WEB/0] OUT JVM Memory Configuration: -Xmx1420872K -Xss1M -XX:ReservedCodeCacheSize=240M -XX:MaxDirectMemorySize=10M -XX:MaxMetaspaceSize=164279K
   2018-12-11T23:20:34.62+0000 [APP/PROC/WEB/0] OUT   .   ____          _            __ _ _
   2018-12-11T23:20:34.62+0000 [APP/PROC/WEB/0] OUT  /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
   2018-12-11T23:20:34.62+0000 [APP/PROC/WEB/0] OUT ( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
   2018-12-11T23:20:34.62+0000 [APP/PROC/WEB/0] OUT  \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
   2018-12-11T23:20:34.62+0000 [APP/PROC/WEB/0] OUT   '  |____| .__|_| |_|_| |_\__, | / / / /
   2018-12-11T23:20:34.62+0000 [APP/PROC/WEB/0] OUT  =========|_|==============|___/=/_/_/_/
   2018-12-11T23:20:34.62+0000 [APP/PROC/WEB/0] OUT  :: Spring Boot ::                        
   2018-12-11T23:20:34.76+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:34.766  INFO 14 --- [           main] pertySourceApplicationContextInitializer : 'cloud' property source added
   2018-12-11T23:20:34.77+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:34.770  INFO 14 --- [           main] nfigurationApplicationContextInitializer : Reconfiguration enabled
   2018-12-11T23:20:34.77+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:34.777  INFO 14 --- [           main] o.s.boot.SpringApplication               : Starting application on dbc47315-0072-402d-5f8d-7970 with PID 14 (started by vcap in /home/vcap/app)
   2018-12-11T23:20:34.77+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:34.777  INFO 14 --- [           main] o.s.boot.SpringApplication               : The following profiles are active: cloud
   2018-12-11T23:20:34.81+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:34.815  INFO 14 --- [           main] s.c.a.AnnotationConfigApplicationContext : Refreshing org.springframework.context.annotation.AnnotationConfigApplicationContext@2d8e6db6: startup date [Tue Dec 11 23:20:34 UTC 2018]; root of context hierarchy
   2018-12-11T23:20:35.67+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:35.678  WARN 14 --- [           main] o.h.v.m.ParameterMessageInterpolator     : HV000184: ParameterMessageInterpolator has been chosen, EL interpolation will not be supported
   2018-12-11T23:20:35.79+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:35.797  INFO 14 --- [           main] com.wavefront.config.BeansConfiguration  : Using reactor.ipc.netty.pool.maxConnections: 32
   2018-12-11T23:20:35.91+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:35.914  INFO 14 --- [           main] ContainerTrustManagerFactory$PKIXFactory : Adding System Trust Manager
   2018-12-11T23:20:35.91+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:35.916  INFO 14 --- [           main] ContainerTrustManagerFactory$PKIXFactory : Adding TrustManager for /etc/ssl/certs/ca-certificates.crt
   2018-12-11T23:20:35.91+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:35.919  INFO 14 --- [tificates.crt-0] org.cloudfoundry.security.FileWatcher    : Start watching /etc/ssl/certs/ca-certificates.crt
   2018-12-11T23:20:36.05+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:36.051  INFO 14 --- [           main] c.s.FileWatchingX509ExtendedTrustManager : Initialized TrustManager for /etc/ssl/certs/ca-certificates.crt
   2018-12-11T23:20:36.06+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:36.063  INFO 14 --- [           main] ContainerTrustManagerFactory$PKIXFactory : Adding System Trust Manager
   2018-12-11T23:20:36.16+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:36.163  INFO 14 --- [           main] ContainerTrustManagerFactory$PKIXFactory : Adding System Trust Manager
   2018-12-11T23:20:36.34+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:36.342  INFO 14 --- [           main] com.wavefront.proxy.ProxyForwarderImpl   : Forwarding PCF metrics to Wavefront proxy at 10.135.148.171:2878
   2018-12-11T23:20:36.74+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:36.745  WARN 14 --- [           main] o.h.v.m.ParameterMessageInterpolator     : HV000184: ParameterMessageInterpolator has been chosen, EL interpolation will not be supported
   2018-12-11T23:20:37.02+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.022  INFO 14 --- [           main] o.s.j.e.a.AnnotationMBeanExporter        : Registering beans for JMX exposure on startup
   2018-12-11T23:20:37.02+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.027  INFO 14 --- [           main] o.s.c.support.DefaultLifecycleProcessor  : Starting beans in phase 0
   2018-12-11T23:20:37.04+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.047 DEBUG 14 --- [ry-client-nio-2] cloudfoundry-client.response             : 200    https://api.sys.<snipped>:443 (427 ms)
   2018-12-11T23:20:37.07+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.071 DEBUG 14 --- [ry-client-nio-2] cloudfoundry-client.request              : GET    https://api.sys.<snipped>:443/v2/info
   2018-12-11T23:20:37.08+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.086 DEBUG 14 --- 
   2018-12-11T23:20:37.09+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.089  INFO 14 --- [           main] .s.FirehoseToWavefrontProxyConnectorImpl : Connecting to firehose using subscription id: firehose-to-wavefront-proxy and forwarding following event types: [COUNTER_EVENT, VALUE_METRIC, CONTAINER_METRIC] in parallel: 4
   2018-12-11T23:20:37.10+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.103 DEBUG 14 --- [rehose-nozzle-1] cloudfoundry-client.request              : GET    https://api.sys.<snipped>:443
   2018-12-11T23:20:37.10+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.104  INFO 14 --- [           main] o.s.boot.SpringApplication               : Started application in 2.791 seconds (JVM running for 4.211)
   2018-12-11T23:20:37.26+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.263 DEBUG 14 --- [ry-client-nio-4] cloudfoundry-client.response             : 200    https://api.sys.<snipped>:443 (176 ms)
   2018-12-11T23:20:37.27+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.279 DEBUG 14 --- [ry-client-nio-5] cloudfoundry-client.response             : 200    https://api.sys.<snipped>:443 (176 ms)
   2018-12-11T23:20:37.28+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.281 DEBUG 14 --- [ry-client-nio-4] cloudfoundry-client.request              : GET    https://api.sys.<snipped>:443/v2/info
   2018-12-11T23:20:37.29+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.290 DEBUG 14 --- [ry-client-nio-5] cloudfoundry-client.request              : WS     https://doppler.sys.<snipped>:443/firehose/firehose-to-wavefront-proxy
   2018-12-11T23:20:37.46+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.461 DEBUG 14 --- [ry-client-nio-6] cloudfoundry-client.response             : 200    https://api.sys.<snipped>:443/v2/info (180 ms)
   2018-12-11T23:20:37.46+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:37.464 DEBUG 14 --- [ry-client-nio-6] cloudfoundry-client.request              : POST   https://login.sys.<snipped>:443/oauth/token
   2018-12-11T23:20:38.03+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:38.038 DEBUG 14 --- [ry-client-nio-3] cloudfoundry-client.response             : 200    https://api.sys.<snipped>:443/v2/info (966 ms)
   2018-12-11T23:20:38.06+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:38.067  WARN 14 --- [ry-client-nio-3] cloudfoundry-client.compatibility        : Client supports API version 2.112.0 and is connected to server with API version 2.98.0. Things may not work as expected.
   2018-12-11T23:20:38.10+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:38.104 DEBUG 14 --- [ry-client-nio-8] cloudfoundry-client.response             : 101    https://doppler.sys.<snipped>:443/firehose/firehose-to-wavefront-proxy (814 ms)
   2018-12-11T23:20:38.18+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:38.180  INFO 14 --- [ry-client-nio-8] com.wavefront.proxy.ProxyForwarderImpl   : Total number of metrics sent: 1
   2018-12-11T23:20:42.18+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:42.183  INFO 14 --- [ry-client-nio-8] com.wavefront.proxy.ProxyForwarderImpl   : Total number of metrics sent: 4826
   2018-12-11T23:20:47.18+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:47.180  INFO 14 --- [ry-client-nio-8] com.wavefront.proxy.ProxyForwarderImpl   : Total number of metrics sent: 8194
   2018-12-11T23:20:52.19+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:52.195  INFO 14 --- [ry-client-nio-8] com.wavefront.proxy.ProxyForwarderImpl   : Total number of metrics sent: 12241
   2018-12-11T23:20:57.19+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:20:57.189  INFO 14 --- [ry-client-nio-8] com.wavefront.proxy.ProxyForwarderImpl   : Total number of metrics sent: 16849
   2018-12-11T23:21:02.18+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:21:02.186  INFO 14 --- [ry-client-nio-8] com.wavefront.proxy.ProxyForwarderImpl   : Total number of metrics sent: 22801
   2018-12-11T23:21:07.22+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:21:07.227  INFO 14 --- [ry-client-nio-8] com.wavefront.proxy.ProxyForwarderImpl   : Total number of metrics sent: 26524
   2018-12-11T23:21:12.18+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:21:12.188  INFO 14 --- [ry-client-nio-8] com.wavefront.proxy.ProxyForwarderImpl   : Total number of metrics sent: 31944
   2018-12-11T23:21:17.19+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:21:17.197  INFO 14 --- [ry-client-nio-8] com.wavefront.proxy.ProxyForwarderImpl   : Total number of metrics sent: 37102
   2018-12-11T23:21:22.18+0000 [APP/PROC/WEB/0] OUT 2018-12-11 23:21:22.188  INFO 14 --- [ry-client-nio-8] com.wavefront.proxy.ProxyForwarderImpl   : Total number of metrics sent: 44185
```

After the last log message above, the nozzle stops receiving any data. Looking to see if folks here have any ideas on how we can debug this issue further.

This is where we subscribe to the firehose:

https://github.com/wavefrontHQ/cloud-foundry-nozzle/blob/8c55258d3778b89e3c0838624f71ab8e19eb1a4e/src/main/java/com/wavefront/service/FirehoseToWavefrontProxyConnectorImpl.java#L74

**Comment by veeragollababu** [comment]
I am also facing the same kind of issue. Did you get any solution for this problem?

**Comment by dmikusa-pivotal** [comment]
This is a very old issue so I'm going to close this out. If you are experiencing this or a similar issue using the latest 4.x or 5.x release of the cf-java-client, then please open a new issue and include the [suggested troubleshooting information](https://github.com/cloudfoundry/cf-java-client#troubleshooting). Thanks!
